lib/client.js

```diff
@@ -358,68 +367,56 @@ Client.prototype.execute = function () {
  * <p>
  *   When dealing with paged results, [ResultSet#nextPage()]{@link module:types~ResultSet#nextPage} method can be used
  *   to retrieve the following page. In that case, <code>rowCallback()</code> will be again called for each row and
- *   this callback will be invoked when all rows in the following page has been retrieved.
+ *   the final callback will be invoked when all rows in the following page has been retrieved.
  * </p>
+ * @example <caption>Using per-row callback and arrow functions</caption>
+ * client.eachRow(query, params, { prepare: true }, (n, row) => console.log(n, row), err => console.error(err));
+ * @example <caption>Overloads</caption>
+ * client.eachRow(query, rowCallback);
```

Comment:
The other methods only support left-to-write optional arguments. 
For example: In `#execute(query, params, options, callback)`
`params` and `options` are optional. The second optional parameter (`options`) can only be defined if the first is defined (`params`).